### PR TITLE
engine: only override write stall configurations if unspecified (#12127)

### LINKED ISSUE #13467: Support configuring RocksDB write-stall settings smaller than flow control
## Bug Report

Related to #11424

Support using smaller config for RocksDB so that compaction can be triggered more often.

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #12127 to release-5.4
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tikv repo:
git pr https://github.com/tikv/tikv/pull/12735
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tikv.git pr/12735:release-5.4-2a508a583c52
```

---

Signed-off-by: tabokie <xy.tao@outlook.com>

### What is changed and how it works?

Issue Number: Close #13467, Ref #11424

What's Changed:

Follow up on #11840.

This is handy if we want to make RocksDB compaction a bit faster without triggering flow control.

### Related changes

- Doc change

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -4666,4 +4733,155 @@ mod tests {
...
+            (
+                "raftstore.raft_store_max_leader_lease",
+                "raft_store.raft_store_max_leader_lease",
+            ),
+        ];
+
+        for (name, res) in cases {
+            assert_eq!(serde_to_online_config(name.into()).as_str(), res);
+        }
+    }
```
Comment (tabokie) [inline src/config.rs:4799]:
```suggestion
```

File: src/config.rs [hunk 2]
```diff
@@ -4666,4 +4733,155 @@ mod tests {
...
+        assert_eq!(
+            cfg.rocksdb.defaultcf.level0_slowdown_writes_trigger,
+            Some(1)
+        );
+        assert_eq!(
+            cfg.rocksdb.defaultcf.soft_pending_compaction_bytes_limit,
+            Some(ReadableSize::gb(1))
+        );
+    }
+>>>>>>> 2a508a583... engine: only override write stall configurations if unspecified (#12127)
```
Comment (tabokie) [inline src/config.rs:4886]:
```suggestion
```

**Review by tabokie — APPROVED** [review]
Waiting on #12738.

**Review by zhangjinpeng1987 — APPROVED** [review]
LGTM

**Pull request merged**